api/v1beta1/cluster_types.go

```diff
@@ -279,8 +279,9 @@ func (v APIEndpoint) String() string {
 // +kubebuilder:resource:path=clusters,shortName=cl,scope=Namespaced,categories=cluster-api
 // +kubebuilder:storageversion
 // +kubebuilder:subresource:status
-// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp",description="Time duration since creation of Cluster"
 // +kubebuilder:printcolumn:name="Phase",type="string",JSONPath=".status.phase",description="Cluster status such as Pending/Provisioning/Provisioned/Deleting/Failed"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp",description="Time duration since creation of Cluster"
+// +kubebuilder:printcolumn:name="Version",type="string",JSONPath=".spec.topology.version",description="Kubernetes version associated with this Cluster"
```

Comment:
This might be confusing intent vs real state of the world. Is there a Version in the status we can use? if not, should we add that to the status?

Reply:
Hm, good point. I think it's hard to say what the current version of a Cluster is (do we look at KCP/MachineDeployments/both?).

It's fine for me to drop the version column here for now.

We have the same problem in KCP. Where we print the `.spec.version` as `VERSION` column independently of if KCP is already entirely reconciled.

Reply:
>Hm, good point. I think it's hard to say what the current version of a Cluster is (do we look at KCP/MachineDeployments/both?).

Not that need it to solve it now, but at minimum I think it'd be valuable to have a status field for the control plane current version. This should be the source of truth for any consumer, e.g a Node that does not want break the kube version skew policy.
Then consider if there's any value in having a semantic that represents the overall cluster version.

Reply:
Yup, we currently only have a .status.version on ControlPlane resources (for ControlPlane implementations using versions)

https://cluster-api.sigs.k8s.io/developer/architecture/controllers/control-plane.html#required-status-fields-for-implementations-using-version

Reply:
thanks! may be can consider to bubble that up to the topology and print that, just a thought.

Reply:
Yup, definitely an idea, but let's move it to an issue. I think it could be controversial as control planes don't have to support versions (although they have to if the should be compatible to ClusterClass)